garden_ai/modal/classes.py

```diff
@@ -0,0 +1,56 @@
+from typing import TYPE_CHECKING, TypeVar, Any
+from garden_ai.modal.functions import ModalFunction
+
+from ..schemas.modal import ModalFunctionMetadata
+
+
+if TYPE_CHECKING:
+    from garden_ai.client import GardenClient
+else:
+    GardenClient = TypeVar("GardenClient")
+
+
+class ModalClassWrapper:
+    """
+    A wrapper for Modal class objects that allows method access and invocation through Garden.
+    """
+
+    def __init__(
+        self,
+        class_name: str,
+        methods: list[ModalFunction],
+        client: GardenClient | None = None,
+    ):
+        self.class_name = class_name
+        # Create method lookup internally
+        self._methods = {
+            method.metadata.function_name.split(".")[-1]: method for method in methods
+        }
+        self.client = client
```

Comment:
would instances of this really need a client to `__init__`? since calling a modal class will resolve to a `ModalFunction` attribute (which already has a client) anyways, I think the only place we really need a client arg is the `from_metadata` method so we can inject one for the `ModalFunction`s